modules/custom/az_card/src/Plugin/Field/FieldFormatter/AZCardDefaultFormatter.php

```diff
@@ -123,6 +123,7 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
 
       // Link.
       $link_render_array = [];
+      $link_url = '';
       if ($item->link_title || $item->link_uri) {
```

Comment:
## Difference Between URI and URL in Drupal

In Drupal, as in general web development, the terms URI (Uniform Resource Identifier) and URL (Uniform Resource Locator) are often used interchangeably, but they have distinct meanings and uses. Here’s an overview of the differences and how they are typically used in Drupal:

### URI (Uniform Resource Identifier)

- **Definition**: A URI is a string of characters used to identify a resource. It is a broader term that includes URLs.
- **Components**: A URI can contain a scheme, authority, path, query, and fragment. 
- **Example**: 
  - URL: `http://example.com/path?query#fragment`
  - File URI: `public://example.txt`

### URL (Uniform Resource Locator)

- **Definition**: A URL is a specific type of URI that provides the means to locate a resource by describing its primary access mechanism (e.g., its network location).
- **Components**: A URL typically includes the scheme (e.g., `http` or `https`), the authority (e.g., `example.com`), and the path to the resource.
- **Example**: `http://example.com/path?query#fragment`

### How They Are Used in Drupal

#### URI in Drupal
- **File URIs**: Drupal often uses URIs to refer to files. For example, `public://example.txt` or `private://example.txt`. These URIs are not URLs, but they identify resources in a way that Drupal understands and can manage.
- **Entity URIs**: Entities in Drupal, such as nodes, users, and terms, have URIs that identify them within the system. For example, `entity:node/1` refers to the node with ID 1.

#### URL in Drupal
- **Path Aliases**: Drupal uses URLs to provide human-readable paths to resources. For example, a node might have a URL like `http://example.com/node/1` or, more commonly with path aliases, `http://example.com/about-us`.
- **Linking**: URLs are used to create hyperlinks in content, menus, and other user interfaces. They point to specific locations on the web where resources can be accessed.

### Practical Examples

#### Using URIs
- **File System**: When managing files, you might see URIs like `public://myfile.txt`. Drupal knows to look in the public files directory for `myfile.txt`.
- **Entity References**: When programmatically referencing an entity, you might use URIs such as `entity:node/1`.

#### Using URLs
- **Linking to Content**: When you create a link to a piece of content, you use a URL like `/node/1` or `/about-us`.
- **Site Configuration**: In site settings, you might configure the base URL of your site, which Drupal uses to generate absolute URLs.

### Summary

- **URI**: A broader term used to identify a resource, which can be a URL or other identifier. In Drupal, URIs are often used for file paths and entity references.
- **URL**: A specific type of URI that provides the location of a resource. In Drupal, URLs are used for web addresses, linking, and navigation.

Understanding the distinction and appropriate use of URIs and URLs can help in managing resources and creating links within a Drupal site effectively.